bolt/lib/Core/MCPlusBuilder.cpp

```diff
@@ -147,6 +147,92 @@ bool MCPlusBuilder::isTailCall(const MCInst &Inst) const {
   return false;
 }
 
+void MCPlusBuilder::setNegateRAState(MCInst &Inst) const {
+  assert(!hasAnnotation(Inst, MCAnnotation::kNegateState));
+  setAnnotationOpValue(Inst, MCAnnotation::kNegateState, true);
+}
+
+bool MCPlusBuilder::hasNegateRAState(const MCInst &Inst) const {
+  if (hasAnnotation(Inst, MCAnnotation::kNegateState))
+    return true;
+  return false;
```

Comment:
nit (could repeat for all has/is introduced methods below, excl. `isRAStateUnknown`)

```suggestion
  return hasAnnotation(Inst, MCAnnotation::kNegateState);
```

Reply:
yupp, `isRAStateUnknown` would be a bit confusing that way